xdsl/dialects/arith.py

```diff
@@ -767,21 +769,25 @@ def print(self, printer: Printer):
 @irdl_op_definition
 class Addf(FloatingPointLikeBinaryOp):
     name = "arith.addf"
+    traits = frozenset((Pure(),))
```

Comment:
Can we not just add the trait on the parent op? So on FloatingPointLikeBinaryOp in this case.